### PR TITLE
DrivAerML: dropout regularization sweep at champion config

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The DrivAerML champion uses NO explicit regularization (no WD, no gc, no dropout, no EMA). Dropout is the most standard neural network regularization — randomly zeroing activations during training to prevent co-adaptation of features. With only 400 training cases and a 4L/512d model (~4M parameters), overfitting is plausible.

Dropout has never been tested on DrivAerML in this programme. Small dropout (0.05-0.1) applied to the attention/FFN layers could improve generalization on the 34-case validation split without disrupting the champion's gradient dynamics.

## Instructions

Add dropout to the model's attention and FFN layers (apply after each attention block and after each FFN block, before residual connection). Run two values:

**Run 1: dropout=0.05**
```bash
cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --dropout 0.05 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group dm-dropout
```

**Run 2: dropout=0.1**
```bash
cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --dropout 0.1 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group dm-dropout
```

If train.py doesn't have a --dropout flag, add one. Apply `nn.Dropout(p)` after each attention output and each FFN output, before the residual add. Set p=0 during eval (standard PyTorch behavior with model.eval()). No --grad-clip, no --weight-decay. Report best val_primary/surface_rel_l2_pct.

## Baseline

- **val_primary/surface_rel_l2_pct:** 3.997% at epoch 467
- **Config:** 4L/512d/8H, AdamW lr=5e-4, T_max=30, no-EMA, Fourier, no gc, no WD, **no dropout**
- **W&B run:** bht6h42t
- **Reproduce:** `cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3121 because ADVISOR: Closing — 3 days without results, student agent appears stuck. Dropout regularization at old T_max=30 config is superseded by T_max=36+lr=4.8e-4 direction. Reassigning..
